### PR TITLE
MINIFI-225: Add Linter for Google style guide

### PR DESCRIPTION
**Pull request by phrocker**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by phrocker** [comment]
Please provide feedback for this approach. I will update the readme to make it a bit clear if more is needed. I kept it intentionally short until we can make this a required process.

**Comment by apiri** [comment]
reviewing

File: thirdparty/google-styleguide/cpplint.py [hunk 1]
```diff
@@ -0,0 +1,6154 @@
+#!/usr/bin/env python
```
Comment (apiri) [inline thirdparty/google-styleguide/cpplint.py]: We will need to capture this in our LICENSE file

File: thirdparty/google-styleguide/cppguide.html [hunk 2]
```diff
@@ -0,0 +1,6170 @@
+<!DOCTYPE html>
```
Comment (apiri) [inline thirdparty/google-styleguide/cppguide.html]: Should likely just link to this in docs some place.  As best as I can tell, this file is CC-By 3.0 as per https://github.com/google/styleguide/blob/gh-pages/README.md.  This falls under a Category B (http://www.apache.org/legal/resolved.html#category-b) which seems to have some caveats best avoided unless there is a compelling reason to include

File: thirdparty/google-styleguide/eclipse-cpp-google-style.xml [hunk 3]
```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (apiri) [inline thirdparty/google-styleguide/eclipse-cpp-google-style.xml]: Might also want to just link to this as per the LICENSE concerns above.

File: thirdparty/google-styleguide/cppguide.html (hunk 2 above)
Comment (phrocker) [inline thirdparty/google-styleguide/cppguide.html]: Agreed. Will change this.

**Comment by apiri** [comment]
reviewing updates

**Comment by apiri** [comment]
@phrocker thanks for the updates, the changes look good and should cover us on the license front.  will get this merged.

**Pull request closed without merging**